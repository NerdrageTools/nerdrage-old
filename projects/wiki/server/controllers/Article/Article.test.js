import { MongoClient } from 'mongodb'

describe('controllers/Article', () => {
  const campaign = 'qa'
  let connection
  let db

  beforeAll(async () => {
    connection = await MongoClient.connect(global.MONGO_URI, { useNewUrlParser: true })
    db = await connection.db(global.MONGO_DB_NAME)
  })
  afterAll(async () => {
    await connection.close()
    await db.close()
  })

  it('mocks db', async () => {
    expect(await db.collection('articles').findOne({ campaign, slug: 'foo' })).toMatchSnapshot()
  })
})
