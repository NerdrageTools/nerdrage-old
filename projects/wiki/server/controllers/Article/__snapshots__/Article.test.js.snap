// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controllers/Article mocks db 1`] = `
Object {
  "_id": "foo",
  "aliases": Array [
    "foo",
    "few",
    "fooh",
  ],
  "campaign": "qa",
  "data": Object {
    "dummy": "data",
  },
  "html": "Foo <b>Bar</b>",
  "slug": "foo",
  "tags": Array [
    "test",
  ],
  "title": "Foo",
}
`;
